recipes/geographiclib/all/conanfile.py

```diff
@@ -0,0 +1,59 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class GeographiclibConan(ConanFile):
+    name = "geographiclib"
+    description = "Convert geographic units and solve geodesic problems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://geographiclib.sourceforge.io"
+    license = "MIT"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            self.options.remove("fPIC")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "GeographicLib-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions['GEOGRAPHICLIB_LIB_TYPE'] = 'SHARED' if self.options.shared else 'STATIC'
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        # it does not work on Windows but is not needed
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "add_subdirectory (js)", "")
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        for folder in ["share", os.path.join("lib", "python"), os.path.join("lib", "pkgconfig"), os.path.join("lib", "cmake"), "sbin", "python", "matlab", "doc", "cmake"]:
+            tools.rmdir(os.path.join(os.path.join(self.package_folder, folder)))
+        if self.settings.os == "Linux":
+            tools.rmdir(os.path.join(os.path.join(self.package_folder, "bin")))
+        elif self.settings.os == "Windows":
+            for item in os.listdir(os.path.join(os.path.join(self.package_folder, "bin"))):
+                if not item.startswith("Geographic"):
```

Comment:
```suggestion
                if not item.startswith("Geographic") or item.endswith(".pdb"):
```

Reply:
delete pdb files too

Reply:
I don't have access to my PC now, I'll fix it later. Or if you can commit that fix feel free to do ir.

Reply:
You can also patch the sources and remove/disable
```
if (MSVC AND GEOGRAPHICLIB_SHARED_LIB)
  install (FILES $<TARGET_PDB_FILE:${PROJECT_SHARED_LIBRARIES}>
    DESTINATION bin OPTIONAL)
endif ()
```
from `self._source_subfolder/src/CMakeLists.txt`

Reply:
I think is easier to just remove the files. Is cleaner since we are removing a bunch of files already. Thanks for the help though!